src/python/pants/backend/scala/dependency_inference/ScalaParser.scala

```diff
@@ -237,23 +239,19 @@ class SourceAnalysisTraverser extends Traverser {
       )
     }
 
-    case Defn.Class(mods, nameNode, tparams, ctor, templ) => {
-      visitMods(mods)
-      val name = extractName(nameNode)
+    case defn: Member.Type with WithMods with WithTParamClause with WithCtor with WithTemplate => // traits, enums and classes
+      visitMods(defn.mods)
+      val name = extractName(defn.name)
       recordProvidedName(name, sawClass = true)
-      apply(tparams)
-      apply(ctor)
-      visitTemplate(templ, name)
-    }
-
-    case Defn.Trait(mods, nameNode, tparams, ctor, templ) => {
+      apply(defn.tparamClause)
+      apply(defn.ctor)
+      visitTemplate(defn.templ, name)
+    case Defn.EnumCase.After_4_6_0(mods, nameNode, tparamClause, ctor, _) =>
       visitMods(mods)
       val name = extractName(nameNode)
-      recordProvidedName(name, sawTrait = true)
```

Comment:
Is `sawTrait` no longer relevant? Is the value used anywhere?

Reply:
Thanks @tdyas, that is an excellent question! The answer is that when I saw this code, I decided to refactor this code because it had been copy-pasted for the `Class` and `Trait` types and I didn't want to copy it _again_ for `Enum`. What I missed is that the code was different here. Now I need to figure out what these flags are actually for.

Reply:
To clarify, when I wrote this code originally, I believe that I had some intent for `sawClass` and `sawTrait`. I believe `sawObject` is still used but the other two may not be. I have forgotten what the reasons are though. :( So if you could audit the code to see if these are still needed, that would be appreciated!

Reply:
I've read the code and afaics they are unused. So I've added another commit to remove them. Thanks!

Reply:
This is actually kinda awesome because it makes this PR one of the best kind: the one were more code is removed than added 🤩